src/coord/src/coord.rs

```diff
@@ -2757,6 +2831,11 @@ where
                 ..
             } = txn
             {
+                // If we need a timestamp but none is available, enqueue ourselves.
+                if !matches!(self.last_open_local_ts, TimestampStatus::Advanced(_)) {
```

Comment:
slight preference for `if matches!(self.last_open_local_ts, TimestampStatus::DidNotAdvance(_)) {`